app/models/manageiq/providers/google/cloud_manager/provision_workflow.rb

```diff
@@ -0,0 +1,23 @@
+class ManageIQ::Providers::Google::CloudManager::ProvisionWorkflow < ::MiqProvisionCloudWorkflow
+  def allowed_instance_types(_options = {})
+    source = load_ar_obj(get_source_vm)
+    ems = get_targets_for_ems(source, :cloud_filter, Flavor, 'flavors')
+    ems.each_with_object({}) { |f, h| h[f.id] = display_name_for_name_description(f) }
+  end
+
+  def allowed_disk_sizes(_options = {})
+    min = 10
+    max = 10 * 1024 # 10 TB
+    (min..max).each_with_object({}) { |i, h| h[i] = i.to_s }
```

Comment:
You can use either approach here:
1. Provide a list of small list of values a user would pick from which can be modified by updating the dialog. For example [miq_dialogs/miq_provision_vmware_dialogs_template.yaml](https://github.com/ManageIQ/manageiq/blob/master/product/dialogs/miq_dialogs/miq_provision_vmware_dialogs_template.yaml#L726)

Or 

2. Change this field to an edit box and create a validation method similar to `validate_regex` which could be used to check the value and warn the user.